api/models.py

```diff
@@ -12,3 +12,31 @@
 def create_auth_token(sender, instance=None, created=False, **kwargs):
     if created:
         Token.objects.create(user=instance)
+
+
+class PurchaseTransactionLogEntry(models.Model):
+    """
+    Tracks economic transactions coming from the X-App API calls for a given user. Intended
+    to be used for debugging and auditing purposes.
+    """
+
+    class Meta:
+        verbose_name = "Purchase transaction log entry"
+        verbose_name_plural = "Purchase transaction log entries"
+
+    class TransactionSourceOptions(models.TextChoices):
```

Comment:
hwat

Reply:
So we can filter on API transactions from XApp and those from stilletime etc. Missing some options because logic for the others are not a priority